components/dashboard/src/prebuilds/list/PrebuildList.tsx

```diff
@@ -0,0 +1,211 @@
+    const hasFilters = !!searchTermDebounced || !!filter.status;
+
+    // Show the table once we're done loading and either have results, or have filters applied
+    const showTable = !isLoading && (prebuilds.length > 0 || hasFilters);
+
+    return (
+        <>
+            <div className="app-container mb-8">
+                <PageHeading title="Prebuilds" subtitle="Review prebuilds of your imported repositories." />
+
+                {isLoading && <LoadingState />}
+
+                {showTable && (
+                    <PrebuildsTable
+                        searchTerm={searchTerm}
+                        prebuilds={prebuilds}
+                        // we check isPreviousData too so we don't show spinner if it's a background refresh
+                        isSearching={isFetching && isPreviousData}
+                        isFetchingNextPage={isFetchingNextPage}
+                        hasNextPage={!!hasNextPage}
+                        filter={filter}
+                        sort={sort}
+                        hasMoreThanOnePage={hasMoreThanOnePage}
+                        onLoadNextPage={() => fetchNextPage()}
+                        onFilterChange={handleFilterChange}
+                        onSearchTermChange={setSearchTerm}
+                        onSort={handleSort}
+                    />
+                )}
+
+                {!showTable && !isLoading && <PrebuildListEmptyState />}
+                {isError && <PrebuildListErrorState error={error} />}
+            </div>
+        </>
+    );
+};
+
+const toApiStatus = (status: StatusOption): ListOrganizationPrebuildsRequest_Filter_State | undefined => {
+    switch (status) {
+        case "failed":
+            return ListOrganizationPrebuildsRequest_Filter_State.FAILED; // todo: adjust to needs of proper status
+        case "succeeded":
+            return ListOrganizationPrebuildsRequest_Filter_State.SUCCEEDED;
+        case "unfinished":
+            return ListOrganizationPrebuildsRequest_Filter_State.UNFINISHED;
+    }
+
+    return undefined;
+};
+
+const parseStatus = (params: URLSearchParams): StatusOption => {
+    const filter = params.get("prebuilds");
+    const validValues = Object.values(STATUS_FILTER_VALUES).filter((val) => !!val);
+    if (filter && validValues.includes(filter as any)) {
+        return filter as StatusOption;
+    }
+
+    return undefined;
+};
+
+const parseSortOrder = (params: URLSearchParams): TableSortOrder => {
+    const sortOrder = params.get("sortOrder");
+    if (sortOrder === "asc" || sortOrder === "desc") {
+        return sortOrder;
+    }
+    return "desc";
+};
+
+const parseSortBy = (params: URLSearchParams): SortField => {
+    const sortBy = params.get("sortBy");
+
+    // todo: potentially allow more fields
+    if (sortBy === "creationTime") {
+        return sortBy;
+    }
+    return "creationTime";
+};
+
+const parseConfigurationId = (params: URLSearchParams): string | undefined => {
```

Comment:
[nit+followup]Will we show something different if configurationId is defined?

Page description `Review prebuilds of your imported repositories.`

Reply:
We most likely should, but it should not be too important in the scope of the PR. When we introduce a way to actually get this state in the UI, we can make sure it looks pretty.